tests/utils.go

```diff
@@ -329,17 +329,6 @@ func GetRunningPodByVirtualMachineInstance(vmi *v1.VirtualMachineInstance, names
 	return pod
 }
 
-func GetPodByVirtualMachineInstance(vmi *v1.VirtualMachineInstance) *k8sv1.Pod {
```

Comment:
This implementation seems to be more efficient then the other one, and it is not related to the flakes.

I think you should introduce it in a different PR and argue why it should be dropped.
It does indeed the same thing, but in a different way. So you need to explain why this one needs to be dropped and not the other.

Reply:
Done, will remove one of the functions in a follow up PR.